ecosystem/indexer-grpc/indexer-grpc-utils/src/storage_format.rs

```diff
@@ -0,0 +1,513 @@
+// Copyright © Aptos Foundation
+
+use crate::default_file_storage_format;
+use anyhow::Context;
+use aptos_protos::{indexer::v1::TransactionsInStorage, transaction::v1::Transaction};
+use flate2::read::{GzDecoder, GzEncoder};
+use prost::Message;
+use ripemd::{Digest, Ripemd128};
+use serde::{Deserialize, Serialize};
+use std::{convert::TryFrom, io::Read};
+
+pub const FILE_ENTRY_TRANSACTION_COUNT: u64 = 1000;
+
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, Eq, PartialEq)]
+pub enum StorageFormat {
+    GzipCompressedProto,
+    // Only used for legacy file format.
+    // Use by cache only.
+    Base64UncompressedProto,
+    // Only used for legacy file format.
+    // Use by file store only.
+    JsonBase64UncompressedProto,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct TransactionsLegacyFile {
+    /// The version of the first transaction in the blob.
+    pub starting_version: u64,
+    /// The transactions in the blob.
+    #[serde(rename = "transactions")]
+    pub transactions_in_base64: Vec<String>,
+}
+
+/// FileStoreMetadata is the metadata for the file store.
+/// It's a JSON file with name: metadata.json.
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, PartialEq, Eq)]
+pub struct FileStoreMetadata {
```

Comment:
we can also use the same single struct pattern for FileStoreMetadata now

Reply:
done.